### PR TITLE
Implement methods to hit the league endpoint

### PR DESCRIPTION
**Pull request by pcg79**
You should pull https://github.com/petems/riot_api/pull/11 first.

Froggen has an awful lot of league data. :frog: 

This feature, I think, underscores the need to implement classes instead of using Hashie.  I found some weirdness when testing the feature in irb:

```
>> entries["entries"].first
=> #<Hashie::Rash is_fresh_blood=false is_hot_streak=false is_inactive=false is_veteran=false last_played=0 league_name="Tryndamere's Inquisitors" league_points=74 losses=0 player_or_team_id="25301501" player_or_team_name="Tovvel" queue_type="RANKED_SOLO_5x5" rank="V" tier="DIAMOND" time_until_decay=-1 wins=333>

>> entries.entries.first
=> ["timestamp", 1386868370660]
```

Where all other API call results I could call a method on the Hashie result, here I have to use Array notation.  I have no idea why, I've never used Hashie that much.  And b/c I've never used Hashie that much I don't know how hard/easy it'll be to implement classes.  I've done a little work with Virtus (https://github.com/solnic/virtus) so hopefully they are similar.

### CODE REVIEW

**Comment by petems** [comment]
Yeah as we get to the bigger returns from the API, it's going to be pretty unwieldy to get data out of it :+1:

Hashie can be a bit weird with it's implementations sometimes, and the data doesn't seem super consistant (Some is quoted, some isn't etc...)

It might be easier to use something else, it should be easy enough to switch up the middleware in Faraday, take the JSON output and then feed it into Virtus?

File: spec/vcr/cassettes/RiotApi_API/_league/_by_summoner/should_return_leagues_data_for_summoner.yml [hunk 1]
```diff
@@ -0,0 +1,862 @@
...
+      - XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX
+      transfer-encoding:
+      - chunked
+      connection:
+      - Close
+    body:
+      encoding: ASCII-8BIT
+      string: !binary |-
+        ewogICJURUFNLTNmN2VhNDMwLTNmNGUtMTFlMy1hNTQ5LTc4MmJjYjRjZTYx
+        YSIgOiB7CiAgICAidGltZXN0YW1wIiA6IDEzODY4NjgwMzAwOTAsCiAgICAi
```
Comment (petems) [inline spec/vcr/cassettes/RiotApi_API/_league/_by_summoner/should_return_leagues_data_for_summoner.yml:39]: Hmm, might be an idea to switch the test up to use someone with less LoL history... this string is making Safari crash on my Mac :smiley_cat:

Comment (pcg79) [inline spec/vcr/cassettes/RiotApi_API/_league/_by_summoner/should_return_leagues_data_for_summoner.yml:39]: Ha, ok, I'll find someone else.

Comment (pcg79) [inline spec/vcr/cassettes/RiotApi_API/_league/_by_summoner/should_return_leagues_data_for_summoner.yml:39]: I'm not in EU so I don't have an friends or anything I can use and all the random EU accounts I'm finding online have crazy long data like Froggen.

If you know of anyone that has a small one, I'll take it and use it.

Comment (petems) [inline spec/vcr/cassettes/RiotApi_API/_league/_by_summoner/should_return_leagues_data_for_summoner.yml:39]: Try `38590679`, someone in my recently played, hasnt had many games :+1:

Comment (petems) [inline spec/vcr/cassettes/RiotApi_API/_league/_by_summoner/should_return_leagues_data_for_summoner.yml:39]: Ah, looking it up, it's going to return eevvveeerrryooonnneeee from a league, regardless. Challenger is probably the easiest as it's only 25 people! I'll merge this now :+1:

**Pull request merged**